common/countdown/countdown.go

```diff
@@ -0,0 +1,83 @@
+// A countdown timer that will mostly be used by XDPoS v2 consensus engine
+package countdown
+
+import (
+	"sync"
+	"time"
+
+	"github.com/XinFinOrg/XDPoSChain/log"
+)
+
+type CountdownTimer struct {
+	lock            sync.RWMutex // Protects the Initilised field
+	resetc          chan int
+	quitc           chan chan struct{}
+	initilised      bool
+	timeoutDuration time.Duration
+	// Triggered when the countdown timer timeout for the `timeoutDuration` period, it will pass current timestamp to the callback function
+	OnTimeoutFn func(time time.Time) error
+}
+
+func NewCountDown(duration time.Duration) *CountdownTimer {
+	return &CountdownTimer{
+		resetc:          make(chan int),
+		quitc:           make(chan chan struct{}),
+		initilised:      false,
+		timeoutDuration: duration,
+	}
+}
+
+// Completely stop the countdown timer from running.
+func (t *CountdownTimer) StopTimer() {
+	q := make(chan struct{})
+	t.quitc <- q
+	<-q
+}
+
+// Reset will start the countdown timer if it's already stopped, or simply reset the countdown time back to the defual `duration`
+func (t *CountdownTimer) Reset() {
+	if !t.getInitilisedValue() {
```

Comment:
getInitilisedValue can rename like `haveInitilised`, return true mean yes, have already Initialised.

Reply:
ok, I think `initilised` is even simpler.